metricbeat/module/kubernetes/_meta/kibana/7/dashboard/Metricbeat-kubernetes-overview.json

```diff
@@ -126,6 +126,28 @@
                                         "field": "kubernetes.container.cpu.usage.nanocores",
                                         "id": "0d5c9221-2bf2-11e7-859b-f78b612cde28",
                                         "type": "sum"
+                                    },
+                                    {
+                                        "id": "8b346300-bf95-11ea-a07c-851701f0d645",
+                                        "type": "avg",
+                                        "field": "metricset.period"
```

Comment:
I guess this will only work well if the period is the same for all events?

Reply:
I guess that too. The thing is if the average is calculated only the periods for this specific metricset and not for the periods of all the metricsets. Otherwise it would be an issue since different metricsets is quite common to have different sampling periods. If that's the case we should restrict the scope of this visualisation to the related metricset only.  @dihmandrake could you verify if that's the case?

Reply:
Yes. If have would have different reporting frequencies this would fail. However, I cannot imagine any way around it.
Ideally we would need an average for every pod in the bucket and then sum the averages for all pods, but this would require us to use something like a "sub-query". To the best of my knowledge, Kibana (Elasticsearch?) does not support any kind of sub-queries. Even then we would need to work around the fact that some pods might have been stopped in the bucket and others are might be started. So that resulting sum would need to based on the `metricset.period` or some kind of start & stop time + weights as well. If have an idea on how to implement it I would love to adjust this PR.

@ChrsMark I am not sure if I understand the second part of your comment for the restrictions correctly. The panel is already limited to `event.module:kubernetes AND (metricset.name:container OR metricset.name:state_node)`. So as long as all Metricbeats running the kubernetes module have the same `period` set we should be fine.

Reply:
Thanks @dihmandrake for clarifying this one!

Since we have the scope set by `event.module:kubernetes AND (metricset.name:container OR metricset.name:state_node)` then it's just fine of we miss some corner cases with restarts etc. I will go along and merge it!